### PR TITLE
Updated the THIRD-PARTY file for 1.5.0

### LINKED ISSUE #1518: Update THIRD-PARTY file for Data Prepper 1.5.0
Update `THIRD-PARTY` file.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Updated the THIRD-PARTY file for 1.5.0
 
Updated via:

```
./gradlew generateThirdPartyReport
```

### Issues Resolved

Resolves #1518
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: THIRD-PARTY [hunk 1]
```diff
@@ -902,27 +900,33 @@ POM License: Apache-2.0 - https://www.apache.org/licenses/LICENSE-2.0.txt
 
 --------------------------------------------------------------------------------
 
-20. Group: com.google.errorprone  Name: error_prone_annotations  Version: 2.11.0
+20. Group: com.google.errorprone  Name: error_prone_annotations  Version: 2.10.0
```
Comment (dinujoh) [inline THIRD-PARTY:903]: Is the version downgrade intended ?

File: THIRD-PARTY [hunk 2]
```diff
@@ -902,27 +900,33 @@ POM License: Apache-2.0 - https://www.apache.org/licenses/LICENSE-2.0.txt
...
 
 POM License: Apache 2.0 - http://www.apache.org/licenses/LICENSE-2.0.txt
 
 --------------------------------------------------------------------------------
 
-21. Group: com.google.errorprone  Name: error_prone_annotations  Version: 2.7.1
+21. Group: com.google.errorprone  Name: error_prone_annotations  Version: 2.11.0
```
Comment (dinujoh) [inline THIRD-PARTY:909]: Is this duplicate of line 903 ? How does `./gradlew generateThirdPartyReport` resolve if there are duplicate in this case ?

File: THIRD-PARTY [hunk 3]
```diff
@@ -1,39 +1,37 @@
 
-Dependency License Report for opensearch-data-prepper 1.4.0-SNAPSHOT
+Dependency License Report for opensearch-data-prepper 1.5.0-SNAPSHOT
 
-1. Group: com.amazonaws  Name: aws-java-sdk-acm  Version: 1.12.155
+1. Group: com.aayushatharva.brotli4j  Name: brotli4j  Version: 1.7.1
```
Comment (sshivanii) [inline THIRD-PARTY:4]: This seems like an odd change, is this supposed to be accurate?

Comment (dlvenable) [reply THIRD-PARTY:4]: The order of dependencies is not consistent, so you can't rely on a line-by-line change. 

Also, the `aws-java-sdk-acm` dependency was removed entirely in this version.

**Review by sshivanii — APPROVED** [review]

**Review by dinujoh — APPROVED** [review]

**Pull request merged**